### PR TITLE
Angle field touch fixes

### PR DESCRIPTION
**Pull request by samelhusseini**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

Fixes https://github.com/google/blockly/issues/2642

### Proposed Changes

There were a number of issues with the angle picker on touch devices.
- The primary fix for the bug was that default browser touch events were being handled on the drag surface that were cancelling the pointer event. This is resolved by adding the CSS ``touch-action: none``. More info [here](https://developer.mozilla.org/en-US/docs/Web/CSS/touch-action) and [here](https://docs.w3cub.com/dom_events/pointercancel/)
- on IOS, which uses touch events, bindEvents_ was blocking a click event from bubbling, so it wasn't being closed on a click
- I couldn't find a reason to have the mouse events set on the gauge as well as the circle (drag surface), so I removed that and disabled pointer events on the gauge.
- I added a click handler on the drag surface for touch devices to do a final update on click, as on touch devices, the value is only updated on drag. (added comment).

### Reason for Changes

Fix angle picker on touch devices (IOS and Android).

### Test Coverage

Playground

Tested on:
* Desktop Chrome
* Desktop Chrome with mobile emulator.
* Desktop Safari
* IOS Safari
* Android Chrome

### CODE REVIEW

**Comment by RoboErikG** [comment]
This lgtm, but @rachel-fenichel to double check

File: core/field_angle.js [hunk 1]
```diff
@@ -280,15 +281,16 @@ Blockly.FieldAngle.prototype.dropdownCreate_ = function() {
...
-  // bindEventWithChecks_ allows it to work without a mousedown/touchstart.
+  // change this behaviour.
   this.clickWrapper_ =
-      Blockly.bindEvent_(svg, 'click', this, this.hide_);
-  this.moveWrapper1_ =
-      Blockly.bindEvent_(circle, 'mousemove', this, this.onMouseMove);
-  this.moveWrapper2_ =
-      Blockly.bindEvent_(this.gauge_, 'mousemove', this, this.onMouseMove);
-
+      Blockly.bindEventWithChecks_(svg, 'click', this, this.hide_);
```
Comment (rachel-fenichel) [inline core/field_angle.js]: Does the change to bindEventWithChecks work for the case of changing it on mousemove without a drag?

Comment (samelhusseini) [reply core/field_angle.js]: Yes, when using mouseevents / pointerevents.

Comment (samelhusseini) [reply core/field_angle.js]: But I assume you're talking about the 'mousemove' two lines below? This one just hides on click.

Comment (rachel-fenichel) [reply core/field_angle.js]: That's right.  The comment above references it as well.

**Pull request merged**